app/helpers/contacts_helper.rb

```diff
@@ -9,4 +9,11 @@ def start_a_conversation_link(aspect, contacts_size)
                   title: t("contacts.index.start_a_conversation")
     end
   end
+
+  def gon_load_contact(contact)
+    Gon.preloads[:contacts_collection] ||= []
+    if contact.id.nil? || !Gon.preloads[:contacts_collection].map {|contact| contact[:id] }.include?(contact.id)
```

Comment:
> What's contact.id.nil? supposed to check for, in what situation does that happen?

In [this place](https://github.com/diaspora/diaspora/pull/6864/files#diff-37afe88261f69735dbcf6835afc2718aR4) in case when contact is `new` it presents a hash with `"id":"null"` value. In this case UI is informed that there is no contact for the person and we still have `"person"` field in the preloaded presented hash which is required for UI components to render.